PyStemmusScope/config_io.py

```diff
@@ -124,9 +124,11 @@ def _copy_data(input_dir: Path, config: dict) -> None:
             str(config[folder]), str(input_dir / folder), dirs_exist_ok=True
         )
 
-    # copy input_data.xlsx
+    # copy input_data.xlsx and input_soilLayThick.csv
     shutil.copy(str(config["input_data"]), str(input_dir))
-
+    soilLay_file = Path(str(config["input_data"]).replace("input_data.xlsx", "input_soilLayThick.csv"))
+    if soilLay_file.exists():
+        shutil.copy(soilLay_file, str(input_dir))
```

Comment:
```suggestion
    # copy soil_layers_thickness, this is optional
    if "soil_layers_thickness" in config:
        if Path(config["soil_layers_thickness"]).is_file():
            shutil.copy(str(config["soil_layers_thickness"]), str(input_dir))
        else:
            raise FileNotFoundError(
                "The key `soil_layers_thickness` is provided in the config file, "
                f"but file {config['soil_layers_thickness']} not found. "
                "Remove the key from the config file or provide the file."
            )
```

Reply:
This suggestion avoids hardcoding the filename and file path. If the key `soil_layers_thickness` is provided in the config path and it refers to the csv file, it will be copied to the input directory of the model. We need to document this, see, issue #105.

Reply:
> This suggestion avoids hardcoding the filename and file path. If the key `soil_layers_thickness` is provided in the config path and it refers to the csv file, it will be copied to the input directory of the model. We need to document this, see, issue #105.

To understand correctly, does your suggestion mean that the path to the soil layer thickness file is in the config file?

Reply:
> To understand correctly, does your suggestion mean that the path to the soil layer thickness file need to be included in the config file?

Not required, it is optional. If the key `soil_layers_thickness` is provided in the config path and it refers to the csv file, it will be copied to the input directory of the model. 

> I am asking because the changes I proposed work without the need to add the path to the soil layer thickness file in the config file

The original code indeed works in your case, but the proposed change offers several improvements in terms of code clarity, maintainability, and robustness. The original code implicitly assumes that both `input_data.xlsx` and `input_soilLayThick` are in the same directory. This assumption might not always hold. Whatever they are, the `config_io` will copy them to `InputPath`, ensuring they are ready for the model. The revised code introduces error handling by raising an error with a clear message. In addition, by using a config key `soil_layers_thickness`, this option is not hidden in the source code, but it is exposed to users.

Reply:
There are no more changes needed. Later, we can fix the documentation and explain the keys in the config file in a follow-up PR, see #105